src/Illuminate/Foundation/Testing/Concerns/InteractsWithRedis.php

```diff
@@ -21,93 +24,344 @@ trait InteractsWithRedis
+        }
+
+        if (defined('Redis::SERIALIZER_JSON')) {
+            $connections[] = 'phpredis_serializer_json';
+        }
+
+        if (defined('Redis::SERIALIZER_MSGPACK')) {
+            $connections[] = 'phpredis_serializer_msgpack';
+        }
+
+        if (defined('Redis::COMPRESSION_LZF')) {
+            $connections[] = 'phpredis_compression_lzf';
+        }
+
+        if (defined('Redis::COMPRESSION_ZSTD')) {
+            $connections[] = 'phpredis_compression_zstd';
+            $connections[] = 'phpredis_compression_zstd_default';
+            $connections[] = 'phpredis_compression_zstd_min';
+            $connections[] = 'phpredis_compression_zstd_max';
+        }
+
+        if (defined('Redis::COMPRESSION_LZ4')) {
+            $connections[] = 'phpredis_compression_lz4';
+            $connections[] = 'phpredis_compression_lz4_default';
+            $connections[] = 'phpredis_compression_lz4_min';
+            $connections[] = 'phpredis_compression_lz4_max';
+        }
 
-        foreach ($this->redisDriverProvider() as $driver) {
-            $this->redis[$driver[0]]->connection()->disconnect();
+        if (defined('Redis::SERIALIZER_MSGPACK') && defined('Redis::COMPRESSION_LZ4')) {
+            $connections[] = 'phpredis_msgpack_and_lz4';
         }
+
+        return $connections;
     }
 
     /**
-     * Get redis driver provider.
+     * Data provider for tests that lists a default set of redis connections.
      *
      * @return array
      */
-    public function redisDriverProvider()
+    public function redisConnectionDataProvider()
     {
-        return [
-            ['predis'],
-            ['phpredis'],
-        ];
+        return (new Collection($this->getRedisConnections()))->mapWithKeys(function ($label) {
+            return [
+                $label => [
+                    $label,
+                ],
+            ];
+        })->all();
     }
 
     /**
-     * Run test if redis is available.
+     * Extended data provider for tests that also lists special configurations
+     * like serialization and compression support on phpredis.
      *
-     * @param  callable  $callback
-     * @return void
+     * @return array
      */
-    public function ifRedisAvailable($callback)
+    public function extendedRedisConnectionDataProvider()
     {
-        $this->setUpRedis();
+        return (new Collection($this->getExtendedRedisConnections()))->mapWithKeys(function ($label) {
+            return [
+                $label => [
+                    $label,
+                ],
+            ];
+        })->all();
+    }
 
-        $callback();
+    /**
+     * Initializes a new RedisManager with the given driver and config.
+     *
+     * @param  string  $driver
+     * @param  array  $config
+     * @return \Illuminate\Redis\RedisManager
+     */
+    private function initializeRedisManager($driver, $config)
+    {
+        $app = $this->app ?? new Application();
```

Comment:
Why do we need `new Application()` here?

Reply:
First of all it was like that before my changes and second, the manager requires an app instance.

Reply:
Why we create a new one? Some Integration tests do not use the Laravel test case and there we need to create one as the class prop is not set in that case.